File: src/main/java/de/byjoker/myjfql/console/ConsoleImpl.java
package de.byjoker.myjfql.console;

import de.byjoker.myjfql.exception.ConsoleException;
import org.jetbrains.annotations.NotNull;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Scanner;

public class ConsoleImpl implements Console {

    private Scanner scanner = null;

    @Override
    public void clean() {
        println(null);
    }

    @Override
    public void log(String s) {
        println(s);
    }

    @Override
    public void logInfo(String s) {
        println("INFO: " + s);
    }

    @Override
    public void logWarning(String s) {
        println("WARNING: " + s);
    }

    @Override
    public void logError(String s) {
        println("ERROR: " + s);
    }

    @Override
    public void print(String s) {
        if (s == null) {
            System.out.println();
            return;
        }

        System.out.println("[" + time() + "] " + s);
    }

    @Override
    public void println(String s) {
        if (s == null) {
            System.out.println();
            return;
        }

        System.out.println("[" + time() + "] " + s);
    }

    @Override
    public String readPrompt() {
        try {
            return scanner.nextLine();
        } catch (Exception ignore) {
        }

        return "";
    }

    @Override
    public void bind(@NotNull Object... parameters) {
        scanner = new Scanner(System.in);
    }

    @Override
    public void clear() {
        throw new ConsoleException("The standard java console doesn't offer any method for clearing the screen!");
    }

    private String time() {
        return new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date());
    }

}

File: src/main/java/de/byjoker/myjfql/console/ConsoleImpl.kt
package de.byjoker.myjfql.console

import de.byjoker.myjfql.exception.ConsoleException
import org.jline.reader.Completer

class ConsoleImpl : Console {

    override fun info(message: String) {
        TODO("Not yet implemented")
    }

    override fun error(message: String) {
        TODO("Not yet implemented")
    }

    override fun warn(message: String) {
        TODO("Not yet implemented")
    }

    override fun bind(completer: Completer) {
        TODO("Not yet implemented")
    }

    override fun readPrompt(): String {
        TODO("Not yet implemented")
    }

    override fun clear() {
        throw ConsoleException("The standard java console doesn't offer any method for clearing the screen!")
    }

}
